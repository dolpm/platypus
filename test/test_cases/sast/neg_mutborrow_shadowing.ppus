pipe test |'a| [help: ~&'a int, doop: ~& int ] |> ~&'a int {
  |> help;
}

pipe main [] |> unit {
  mut int t = 4;
  mut int p = 6;

  mut int u = 7;

  ~&int y = ~&t;
  ~&int z = ~&p;
  
  ~& int caterpie = test <| [test <| [~&u, z], y];

  ~&u;

  |> ();
}
