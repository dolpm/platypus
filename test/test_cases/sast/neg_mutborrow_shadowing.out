Fatal error: exception Failure("Mutable reference u cannot be used until its alias caterpie has fallen out of scope.")
